R/defStormsDataset.R

```diff
@@ -241,23 +318,294 @@ checkInputsdefStormsDataset <- function(filename, fields, basin, seasons, unitCo
+#' @param fields named character vecor
+#' @param basin character
+#' @param seasons numeric vector
+#' @param unitConversion named character vector
+#' @param notNamed character for not named storms
+#' @param verbose numeric
+#'
+#' @return list of arrays
+getDataFromCsvFile <- function(filename, sep, fields, basin, seasons, unitConversion, notNamed, verbose){
+  
+  if (verbose) {
+    cat("=== Loading data  ===\nOpen database... ")
+  }
+  
+  if(is.null(sep)){
+    sep = ","
+  }
+  
+  dataBase <- utils::read.csv(file = filename, sep = sep)
+  
+  if (verbose) {
+    cat(filename, "opened\nCollecting data ...\n")
+  }
+  
+  # Extract columns
+  filter <- which(colnames(dataBase) %in% fields)
+  dataBaseFiltered <- dataBase[, filter]
+  
+  # Remove sub header
+  dataBaseFiltered <- dataBase[2:dim(dataBaseFiltered)[1], filter]
+  
+  # Remove notNamed storms
+  filter <- which(!(dataBaseFiltered[,fields["names"]] %in% notNamed))
+  dataBaseFiltered <- dataBaseFiltered[filter,]
+  
+  # Remove NA lon-lat row
+  filter <- which(!is.na(dataBaseFiltered[,fields["lon"]]) & !is.na(dataBaseFiltered[,fields["lat"]]))
+  dataBaseFiltered <- dataBaseFiltered[filter,]
+  
+  # Filter by season
+  filter <- which(as.numeric(dataBaseFiltered[,fields["seasons"]]) >= seasons[1] & as.numeric(dataBaseFiltered[,fields["seasons"]]) <= seasons[2])
+  dataBaseFiltered <- dataBaseFiltered[filter,]
+  
+  # Filter by basin
+  if(!is.null((basin))){
+    filter <- which(dataBaseFiltered[,fields["basin"]] == basin)
+    dataBaseFiltered <- dataBaseFiltered[filter,]
+  }
+  
+  # Get dimensions
+  stormNames <- dataBaseFiltered[,fields["names"]] 
+  stormSeasons <- dataBaseFiltered[,fields["seasons"]]
+  sid <- paste0(stormNames, stormSeasons)
+
+  # Do the following code to replace table function
+  k <- 0
+  actualStorm <- sid[1]
+  countObs <- c()
+  names <- c()
+  for (i in 1:length(sid)){
+    if(sid[i] != actualStorm){
+      countObs <- c(countObs, k)
+      names <- c(names, actualStorm)
+      actualStorm <- sid[i]
+      k <- 1
+    }else{
+      k = k + 1
+    }
+  }
+  countObs <- c(countObs, k)
+  names <- c(names, sid[length(t)])
+  names(countObs) <- names
+  
+  # Do not work anymore without sid field
+  # sid <- dataBaseFiltered[,fields["sid"]]
+  # countObs <- table(sid) # count of observation by storm
+  
+  row <- max(countObs) # maximum number of rows
+  len <- length(unique(sid)) # number of unique storm in csv
+  cumulativeIndex <- cumsum(countObs) # starting points for each storm in csv
+  
+  # Initialize template structure
+  templateArray = array(NaN, dim=c(row,len))
+  
+  # Mandatory fields
+  data <- list(
+    names = array(NaN, dim=len),
+    seasons = array(NaN, dim=len),
+    isotimes = templateArray,
+    longitude = templateArray,
+    latitude = templateArray,
+    msw = templateArray
+  )
+  
+  # TODO Remove later
+  if("sshs" %in% names(fields)){
+    data$sshs <- templateArray
+  }
+  
```

Comment:
TBD

Reply:
I meant that it will be removed with the merge of PR 40